src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,649 @@
+                            : [...newOptions, freeformOptionsHeader, newOption];
+                        setOptions(newOptions);
+                        setSelectedOption(newOption);
+                        setSelectedKey(newParsedOptionValue);
+                    }
+                    if (onResourceChange)
+                        onResourceChange(
+                            newParsedOptionValue,
+                            newOptions
+                                ?.filter(
+                                    (option) =>
+                                        option.itemType !==
+                                        SelectableOptionMenuItemType.Header
+                                )
+                                .map((option) => option.data || option.text) ||
+                                []
+                        );
+                } else {
+                    setSelectedOption(existingOption);
+                    setSelectedKey(existingOption.key);
+                    if (onResourceChange)
+                        onResourceChange(
+                            existingOption.data || existingOption.text,
+                            options
+                                ?.filter(
+                                    (option) =>
+                                        option.itemType !==
+                                        SelectableOptionMenuItemType.Header
+                                )
+                                .map((option) => option.data || option.text) ||
+                                []
+                        );
+                }
+            }
+        },
+        [options]
+    );
+
+    const handleOnRemove = useCallback(
+        (option: IComboBoxOption) => {
+            const restOfOptions = options.filter((o) => o.key !== option.key);
+            if (restOfOptions.length) {
+                const lastOption = restOfOptions[restOfOptions.length - 1];
+                if (
+                    lastOption.itemType === SelectableOptionMenuItemType.Header
+                ) {
+                    // remove the '---' freeform options header if there is no option below
+                    restOfOptions.splice(-1);
+                }
+            }
+            setOptions(restOfOptions);
+
+            if (option.key === selectedOption?.key) {
+                setSelectedOption(null);
+                if (onResourceChange)
+                    onResourceChange(
+                        null,
+                        restOfOptions
+                            .filter(
+                                (option) =>
+                                    option.itemType !==
+                                    SelectableOptionMenuItemType.Header
+                            )
+                            .map((option) => option.data || option.text)
+                    );
+            } else {
+                if (onResourceChange)
```

Comment:
can we pull out the notification of the parent outside the conditional since it's duplicated in both cases?

Reply:
Fixed: https://github.com/microsoft/iot-cardboard-js/commit/6a3da6ace5c92b93b9f5386ad828f0e9f3326ae4#diff-3bd141423ed5699ed9c540576ed4472a3ed44ba1f9f6c4e73b720705561b84deR499